core/src/main/java/com/opensymphony/xwork2/conversion/impl/DateConverter.java

```diff
@@ -48,7 +51,28 @@ public Object convertValue(Map<String, Object> context, Object target, Member me
                 }
             } else if (java.util.Date.class == toType) {
                 Date check;
-                DateFormat[] dfs = getDateFormats(locale);
+
+
+                // Add the user provided date format if any
+                final TextProvider tp = findProviderInStack((ValueStack) context.get(ActionContext.VALUE_STACK));
+                SimpleDateFormat globalDateFormat = null;
+
+                if (tp != null) {
+                    String globalFormat = tp.getText(org.apache.struts2.components.Date.DATETAG_PROPERTY);
+
+                    // if tp.getText can not find the property then the
+                    // returned string is the same as input =
+                    // DATETAG_PROPERTY
+                    if (globalFormat != null
+                        && !org.apache.struts2.components.Date.DATETAG_PROPERTY.equals(globalFormat)) {
+                        globalDateFormat = new SimpleDateFormat(globalFormat, locale);
+                    }
+                }
```

Comment:
Could you move this whole section into `getDateFormats` to avoid passing the `globalDateFormat` as a parameter? Also it would be good to extract this logic into a protected method to allow override, so the finally it should look like this:

```java
     DateFormat[] dfs = getDateFormats(globalDateFormat, locale);

...

private DateFormat[] getDateFormats(Locale locale) {
    ...
    DateFormat globalDateFormat = getGlobalDateFormat();
    ...
}

protected getGlobalDateFormat() {
     // your code here
}
```